custom/icds_reports/models/aggregate.py

```diff
@@ -1470,3 +1472,26 @@ class Meta(object):
 
     _agg_helper_cls = AwwIncentiveAggregationDistributedHelper
     _agg_atomic = False
+
+
+
+class DashboardUserActivityReport(models.Model, AggregateMixin):
+    """
+    Daily Update table to hold Dashboard users activity information
+    """
+    username = models.TextField(null=True)
```

Comment:
why are all of these fields `null=True`? Under what circumstances could a user be null?

Also django warns against using null on text and char field: https://docs.djangoproject.com/en/1.11/ref/models/fields/#null

> Avoid using null on string-based fields such as CharField and TextField. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” the Django convention is to use the empty string, not NULL. One exception is when a CharField has both unique=True and blank=True set. In this situation, null=True is required to avoid unique constraint violations when saving multiple objects with blank values.

> For both string-based and non-string-based fields, you will also need to set blank=True if you wish to permit empty values in forms, as the null parameter only affects database storage (see blank).

Reply:
`Under what circumstances could a user be null?` Actually None. Made it not-null.